Cohort2/team4/server/src/api/index.ts

```diff
@@ -1,14 +1,135 @@
 import express from 'express';
-
+import Users from '../models/Users';
+import Events from '../models/Events';
 import MessageResponse from '../interfaces/MessageResponse';
+import { ParamsDictionary } from 'express-serve-static-core';
 
 const router = express.Router();
 
 router.get<{}, MessageResponse>('/', (req, res) => {
   res.json({
     message: 'API',
+    data: null,
+  });
+});
+
+router.get<{}, MessageResponse>('/emojis', (req, res) => {
+  res.json({
+    message:  '😀,😳,🙄',
+    data: null,
   });
 });
 
 
+//Test Route Find One
+router.get<ParamsDictionary, MessageResponse>('/users/:id', async (req, res) => {
+  try {
+    const user = await Users.findByPk(req.params.id);
+    if (user) {
+      res.json({ message: 'User found', data: user });
+    } else {
+      res.status(404).json({ message: 'User not found', data: null });
+    }
+  } catch (error) {
+    res.status(500).json({ message: 'Internal server error', data: null });
+  }
+});
+
+//Test Find All
+//Not working
+router.get<{}, MessageResponse>('/users/all', async (req, res) => {
+  try {
+    const allUsers = await Users.findAll();
+    if (allUsers) {
+      res.json({ message: 'Users found', data: allUsers });
+    } else {
+      res.status(404).json({ message: 'User not found', data: null });
+    }
+  } catch (error) {
+    res.status(500).json({ message: 'Internal server error', data: { error } });
+    console.error(error);
+  } 
+});
+
+//Test Route Create
+router.post<{}, MessageResponse>('/users/create', async (req, res) => {
+  const { first_Name, last_Name, email, auth_key } = req.body;
+  
+  try {
+    const newUser = await Users.create({
+      first_name: first_Name,
+      last_name: last_Name,
+      email: email,
+      auth_key: auth_key,
+    });
+    if (newUser) {
+      res.json({ message: 'User created', data: newUser });
+    } else {
+      res.status(404).json({ message: 'User not created', data: null });
+    }
+  } catch (error) {
+    console.log(error)
+    res.status(500).json({ message: 'Internal server error', data: null });
+  }
+});
+
+
+//Test Route to Find One Event
+router.get<ParamsDictionary, MessageResponse>('/events/:id', async (req, res) => {
```

Comment:
So this is a REST approach to the Events, and not tRPC one. The goal was to use tRPC. I get that it was done for testing purposes, but should we remove it in favour of a tRPC API instead?